### PR TITLE
[Experimental] Add remaining x86 vectors: FMA4, AVX2, AVX2128, AVX512F

### PR DESCRIPTION
**Pull request by diaena**
… AVX2128, AVX512F

### CODE REVIEW

File: src/libm-tester/CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,50 @@ if(COMPILER_SUPPORTS_AVX)
...
+
+  list(APPEND IUT_LIST ${TARGET_IUTAVX2128})
+endif(COMPILER_SUPPORTS_AVX2)
+
+if(COMPILER_SUPPORTS_AVX512F)
+  add_executable(${TARGET_IUTAVX512F} iutsimd.c iutsimdmain.c testerutil.c)
+  target_compile_options(${TARGET_IUTAVX512F}
+    PRIVATE ${FLAGS_ENABLE_AVX512F})
+  target_compile_definitions(${TARGET_IUTAVX512F}
+    PRIVATE ENABLE_AVX2=1)
```
Comment (shibatch) [inline src/libm-tester/CMakeLists.txt]: On line 90, ENABLE_AVX2=1 should be ENABLE_AVX512F=1

File: src/libm/CMakeLists.txt [hunk 2]
```diff
@@ -63,6 +63,54 @@ if(COMPILER_SUPPORTS_AVX)
...
+  add_library(${OBJECT_AVX2128} OBJECT ${SIMD_SOURCES} ${HEADER_AVX2128})
+  target_compile_definitions(${OBJECT_AVX2128} PRIVATE ENABLE_AVX2128=1 DORENAME=1)
+  target_compile_options(${OBJECT_AVX2128} PRIVATE ${FLAGS_ENABLE_AVX2})
+
+  list(APPEND TARGET_OBJECTS ${OBJECT_AVX2128})
+  list(APPEND RENAME_HEADERS ${HEADER_AVX2128})
+endif(COMPILER_SUPPORTS_AVX2)
+
+if(COMPILER_SUPPORTS_AVX512F)
+  set(HEADER_AVX512F ${CMAKE_CURRENT_BINARY_DIR}/include/renameavx2.h)
```
Comment (shibatch) [inline src/libm/CMakeLists.txt]: On line 103, renameavx2.h should be renameavx512f.h

**Comment by diaena** [comment]
Sorry, I've merged this accidentally. Reverting now.

**Comment by diaena** [comment]
Moving to: https://github.com/shibatch/sleef/pull/54

**Pull request merged**